source/server/server.cc

```diff
@@ -614,28 +617,29 @@ RunHelper::RunHelper(Instance& instance, const Options& options, Event::Dispatch
       }) {
   // Setup signals.
   if (options.signalHandlingEnabled()) {
-// TODO(Pivotal): Figure out solution to graceful shutdown on Windows. None of these signals exist
-// on Windows.
-#ifndef WIN32
-    sigterm_ = dispatcher.listenForSignal(SIGTERM, [&instance]() {
-      ENVOY_LOG(warn, "caught SIGTERM");
-      instance.shutdown();
-    });
-
-    sigint_ = dispatcher.listenForSignal(SIGINT, [&instance]() {
-      ENVOY_LOG(warn, "caught SIGINT");
-      instance.shutdown();
-    });
-
-    sig_usr_1_ = dispatcher.listenForSignal(SIGUSR1, [&access_log_manager]() {
-      ENVOY_LOG(info, "caught SIGUSR1. Reopening access logs.");
-      access_log_manager.reopen();
-    });
-
-    sig_hup_ = dispatcher.listenForSignal(SIGHUP, []() {
-      ENVOY_LOG(warn, "caught and eating SIGHUP. See documentation for how to hot restart.");
-    });
-#endif
+    if constexpr (platformSupportsSignal(ENVOY_SIGTERM)) {
+      sigterm_ = dispatcher.listenForSignal(ENVOY_SIGTERM, [&instance]() {
+        ENVOY_LOG(warn, "caught ENVOY_SIGTERM");
+        instance.shutdown();
+      });
+    }
+    if constexpr (platformSupportsSignal(ENVOY_SIGINIT)) {
+      sigint_ = dispatcher.listenForSignal(ENVOY_SIGINIT, [&instance]() {
+        ENVOY_LOG(warn, "caught SIGINT");
+        instance.shutdown();
+      });
+    }
```

Comment:
See suggestion for platform functions that return a list of signals to associate with the clean shutdown handler as a possible alternative to ENVOY_SIGTERM and friends.

https://github.com/envoyproxy/envoy/pull/13954#discussion_r521584038

Reply:
Agreed if the relationship is one-to-many then functions are the way to go. SIGTERM would map to multiple Windows events